File: search_algos/linearsearch.rb
#This is a program written in ruby to perform a linear search

def linear_search(array, key)
  if array.index(key).nil?
    print "Number not found!"
  else
    print "#{key} at index #{array.index(key)}"
  end
end

print "Enter number of numbers"
num = $stdin.gets.chomp.to_i
print "Enter the numbers!"

arr = Array.new(50)
for i in 1..num
  arr[i] = $stdin.gets.chomp.to_i
end
print "Enter the number to be found:"
key = $stdin.gets.chomp.to_i
linear_search(arr, key)

File: search_algos/Linear_search.cpp
/* C++ program for implementation of Linear Search */
#include <iostream> 
using namespace std; 
  
int search(int arr[], int n, int x) 
{ 
    int i; 
    for (i = 0; i < n; i++) 
        if (arr[i] == x) 
            return i; 
    return -1; 
} 
  
int main(void) 
{ 
    cout<<"Enter size of array";
    int n;
    cin>>n;
    int arr[n];
    cout<<"Enter Elements of array";
for(int i=0;i<n;i++)
{
    cin>>arr[i];
}
cout<<"Enter the number u want to search";
int x;
cin>>x;
    int result = search(arr, n, x); 
   (result == -1)? cout<<"Element is not present in array" 
                 : cout<<"Element is present at index " <<result; 
   return 0; 
} 